packages/spacecat-shared-data-access/test/it/db.test.js

```diff
@@ -740,4 +754,66 @@ describe('DynamoDB Integration Test', async () => {
     expect(siteCandidate.getSource()).to.be.undefined;
     expect(siteCandidate.getSiteId()).to.be.undefined;
   });
+
+  it('successfully adds a new top page', async () => {
+    const siteId = (await dataAccess.getSites())[0].getId();
+
+    const siteTopPageData = {
+      siteId,
+      url: 'https://example12345.com/page-12345',
+      traffic: 360420000,
+      source: 'rum',
+      geo: 'au',
+      importedAt: new Date().toISOString(),
+    };
+
+    const newSiteTopPage = await dataAccess.addSiteTopPage(siteTopPageData);
+
+    checkSiteTopPage(newSiteTopPage);
+    expect(newSiteTopPage.getSiteId()).to.equal(siteTopPageData.siteId);
+    expect(newSiteTopPage.getTraffic()).to.equal(siteTopPageData.traffic);
+    expect(newSiteTopPage.getSource()).to.equal(siteTopPageData.source);
+    expect(newSiteTopPage.getGeo()).to.equal(siteTopPageData.geo);
+    expect(newSiteTopPage.getImportedAt()).to.equal(siteTopPageData.importedAt);
+
+    const topPages = await dataAccess.getTopPagesForSite(
+      siteTopPageData.siteId,
+      siteTopPageData.source,
+      siteTopPageData.geo,
+    );
+
+    expect(topPages).to.be.an('array').that.has.lengthOf(1);
+    const topPage = topPages[0];
+    checkSiteTopPage(topPage);
+    expect(topPage.getSiteId()).to.equal(siteTopPageData.siteId);
+    expect(topPage.getTraffic()).to.equal(siteTopPageData.traffic);
+    expect(topPage.getSource()).to.equal(siteTopPageData.source);
+    expect(topPage.getGeo()).to.equal(siteTopPageData.geo);
+    expect(topPage.getImportedAt()).to.equal(siteTopPageData.importedAt);
+  });
+
+  it('retrieves top pages for a site from a specific source and geo in descending traffic order', async () => {
+    const siteId = (await dataAccess.getSites())[0].getId();
+
+    const siteTopPages = await dataAccess.getTopPagesForSite(siteId, 'ahrefs', 'global');
```

Comment:
@AndreiAlexandruParaschiv , here's an example of how you'd get top pages for a site for the audit.